src/tir/schedule/concrete_schedule.cc

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "./concrete_schedule.h"
+
+namespace tvm {
+namespace tir {
+
+Schedule Schedule::Concrete(IRModule mod, int debug_mode) {
+  ObjectPtr<ConcreteScheduleNode> n = make_object<ConcreteScheduleNode>();
+  n->state_ = ScheduleState(mod, debug_mode);
+  n->symbol_table_ = {};
+  n->analyzer_ = std::make_unique<arith::Analyzer>();
+  return Schedule(std::move(n));
+}
+
+/******** Copy ********/
+
+/*! \brief Helper class to copy the sref tree */
+class ScheduleCopier {
+  using TSymbolTable = ConcreteScheduleNode::TSymbolTable;
+  template <class K, class V>
+  using UMap = std::unordered_map<K, V>;
+  template <class K, class V>
+  using SMap = std::unordered_map<K, V, ObjectPtrHash, ObjectPtrEqual>;
+
+ public:
+  static void Copy(const ConcreteScheduleNode* self,
+                   ScheduleState* new_state,  //
```

Comment:
This slash is for?

Reply:
Haha for my pathetic aesthetics of c++ code :-)  will get it removed then